src/daemon/daemon.cpp

```diff
@@ -2668,6 +2680,82 @@ catch (const std::exception& e)
     status_promise->set_value(grpc::Status(grpc::StatusCode::INTERNAL, e.what(), ""));
 }
 
+void mp::Daemon::clone(const CloneRequest* request,
+                       grpc::ServerReaderWriterInterface<CloneReply, CloneRequest>* server,
+                       std::promise<grpc::Status>* status_promise)
+{
+    try
+    {
+        config->factory->require_clone_support();
+        mpl::ClientLogger<CloneReply, CloneRequest> logger{mpl::level_from(request->verbosity_level()),
+                                                           *config->logger,
+                                                           server};
+
+        const auto& source_name = request->source_name();
+        const auto [instance_trail, status] = find_instance_and_react(operative_instances,
+                                                                      deleted_instances,
+                                                                      source_name,
+                                                                      require_operative_instances_reaction);
+        if (status.ok())
+        {
+            const std::string destination_name = generate_destination_instance_name_for_clone(*request);
+
+            auto rollback_clean_up_all_resource_of_dest_instance = sg::make_scope_guard(
+                [this, destination_name]() noexcept -> void { release_resources(destination_name); });
```

Comment:
This doesn't clean up `preparing_instances`, blocking future clones from being created by occupying the next generated name; ie `foo-clone1`.

Reply:
Indeed, removing from `preparing_instances` should be added.